src/crimson/os/seastore/object_data_handler.h

```diff
@@ -54,9 +96,18 @@ struct ObjectDataBlock : crimson::os::seastore::LogicalCachedExtent {
   }
 
   void overwrite(extent_len_t offset, bufferlist bl) {
-    auto iter = bl.cbegin();
-    iter.copy(bl.length(), get_bptr().c_str() + offset);
-    delta.push_back({offset, bl.length(), bl});
+    if (overwrites == nullptr &&
+	get_paddr().get_addr_type() == paddr_types_t::RANDOM_BLOCK) {
```

Comment:
Doesn't seem to me specific to RBM, the mechanism in only related to minimizing memory copy.